content/en/docs/Reference/dependency-resolution.md

```diff
@@ -0,0 +1,351 @@
+
+{{<mermaid>}}
+graph TD
+   
+    subgraph Cluster
+      s(Subscription<br /><br />package: foo<br />channel: stable)
+      o(ClusterServiceVersion<br /><br />FooOperator v1.2.0)
+    end
+    
+    subgraph Catalog
+        stable(package: foo<br />channel: stable) --> e2
+        e2(FooOperator v1.2.3) 
+        e3(FooOperator v1.2.2) 
+        e4(FooOperator v1.2.1) 
+        e5(FooOperator v1.2.0) 
+        e6(FooOperator v1.1.0) 
+        e2 --> e3
+        e3 --> e4
+        e3 --> e5
+        e4 --> e5
+        e5 --> e6
+    end
+    
+    classDef foo fill:#8addf2,stroke:#333,stroke-width:4px;
+    classDef fooSelected fill:#8addf2,stroke:green,stroke-width:4px;
+    classDef bar fill:#ffcc26,stroke:#333,stroke-width:4px;
+    classDef baz fill:#ff7452,stroke:#333,stroke-width:4px;
+
+    class s,o foo;
+    class e3,e4 fooSelected;
+    class v,vs bar;
+    
+    linkStyle default fill:none,stroke-width:2px;
+{{</mermaid>}}
+
+Like dependencies, subscription constraints have preference as well. OLM will choose the newest (closest to the head of the channel) operator that satisfies the subscription constraint. In this example, `FooOperator v1.2.2` would be chosen first.
+
+#### Package constraint
+
+Within a namespace, no two operators may come from the same package.
+
+
+#### Operator Minimization
+
+In the case that there is more than one valid solution, the solution that installs the fewest operators is preferred.
+
+## Best Practices
+
+### Depend on APIs
+
+Operators may add or remove apis at any time - always specify an `olm.gvk` dependency on any APIs your operator requires.
+
+### Set a minimum version
+
+The kubernetes [documentation on api changes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#readme) describes what changes are allowed for kube-style operator APIs. These versioning conventions allow an operator to update an API without bumping the `apiVersion`, as long as the api is backwards-compatible.
+
+For operator dependencies, this means that knowing the `apiVersion` of a dependency may not be enough to ensure the dependent operator works as intended. 
+
+For example:
+
+ - `FooOperator v1.0` provides `v1alpha1` of `Foo`
+ - `FooOperator v1.0.1` adds a new field `spec.bar` to `Foo`, but still at `v1alpha1`
+
+Your operator may require the ability to write `spec.bar` into `Foo`. An `olm.gvk` constraint alone is not enough for OLM to determine that you need `FooOperator v1.0.1` and not `FooOperator v1.0`.
+
+Whenever possible, if a specific operator that provides an API is known ahead of time, specify an additional `olm.package` constraint to set a minimum.
+
+### Omit a maximum version, or allow a very wide range
+
+Because operators provide cluster-scoped resources (ApiServices, CRDs), an operator that specifies a small window for a dependency may unnecessarily constrain updates for other consumers of that dependency.
```

Comment:
```suggestion
Because operators provide cluster-scoped resources (APIServices, CRDs), an operator that specifies a small window for a dependency may unnecessarily constrain updates for other consumers of that dependency.
```